### ISSUE TITLE
False positive for `function-whitespace-after` in a `var()`

### ISSUE THREAD

**Issue by DMShamonov** [stylelint/stylelint#6145]
### What steps are needed to reproduce the bug?

If I use variable interpolation inside a `var()`, rule `function-whitespace-after` returns a false positive.

### What Stylelint configuration is needed to reproduce the bug?

```json
{
"extends": [
   "stylelint-config-recommended",
   "stylelint-config-recess-order"
 ],
 "rules" : {
   ...
   "function-whitespace-after": "always"
  }
}
```

### How did you run Stylelint?

On `pre-commit` hook by `lint-staged` - `"**/*.pcss": "stylelint --fix"`

### Which version of Stylelint are you using?

`14.8.5`

### What did you expect to happen?

Nothing

### What actually happened?

`font-family: var(--font-$(style)-font-family);` changes to `font-family: var(--font-$(style) -font-family);`

Adds whitespace after `$(style)`

**Comment by ybiquitous** [comment]
@DMShamonov Thank you for creating the detailed report and using the template.

Unfortunately, we have decided not to update or fix stylistic rules like `function-whitespace-after`. So please consider using a pretty printer like [Prettier](https://prettier.io/) instead.

For details, see also [this document](https://stylelint.io/user-guide/rules/list#stylistic-issues).

By the way, what syntax `$(style)` are you using? It looks different from SCSS... 🤔

**Comment by DMShamonov** [comment]
@ybiquitous thanks for clarification! We are changing this part of our project soon 🙌 

Syntax like `$(style)` adds `postcss-each` plugin 😃

**Comment by ybiquitous** [comment]
> Syntax like `$(style)` adds `postcss-each` plugin 😃

I didn't know that. Thanks for letting me know! 👍🏼
